WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/MainStateHolder.kt

```diff
@@ -180,12 +234,45 @@ class MainStateHolder(
      */
     private fun speakerIsSupported(): Boolean {
         val hasAudioOutputFeature =
-            application.packageManager.hasSystemFeature(PackageManager.FEATURE_AUDIO_OUTPUT)
-        val devices = application.getSystemService<AudioManager>()!!
+            activity.packageManager.hasSystemFeature(PackageManager.FEATURE_AUDIO_OUTPUT)
+        val devices = activity.getSystemService<AudioManager>()!!
             .getDevices(AudioManager.GET_DEVICES_OUTPUTS)
 
         // We can only trust AudioDeviceInfo.TYPE_BUILTIN_SPEAKER if the device advertises FEATURE_AUDIO_OUTPUT
         return devices.any { it.type == AudioDeviceInfo.TYPE_BUILTIN_SPEAKER } && hasAudioOutputFeature ||
             devices.any { it.type == AudioDeviceInfo.TYPE_BLUETOOTH_A2DP }
     }
+
+    companion object {
+        private val MAX_RECORDING_DURATION = Duration.ofSeconds(10)
+        private const val NUMBER_TICKS = 10
+    }
+}
+
+/**
+ * One of the actions the app interprets. This can either be direct user actions (clicking one of the buttons),
+ * or programmatic actions (finishing playback or recording).
+ */
+sealed class AppAction {
+    object Started : AppAction()
+    object MicClicked : AppAction()
+    object PlayClicked : AppAction()
+    object MusicClicked : AppAction()
+    object PermissionResultReturned : AppAction()
+}
+
+/**
+ * The four "resting" states of the application.
+ */
+sealed class AppState {
+    data class Ready(
+        /**
+         * If true, don't animate to this position
+         */
+        val transitionInstantly: Boolean,
+    ) : AppState()
+
+    object PlayingVoice : AppState()
+    object PlayingMusic : AppState()
+    object Recording : AppState()
```

Comment:
As mentioned above, the `recordingProgress` state could be here to centralize all the info